Core/HandlerMgr.py

```diff
@@ -1,42 +1,45 @@
-
-import inspect
-import imp
 import os
 import re
+import imp
+import inspect
+import collections
 
 from DIRAC import S_OK, S_ERROR, rootPath, gLogger
 from DIRAC.Core.Utilities.ObjectLoader import ObjectLoader
 from DIRAC.Core.Utilities.DIRACSingleton import DIRACSingleton
 from DIRAC.ConfigurationSystem.Client.Helpers import CSGlobals
 
 import WebAppDIRAC
+
+from WebAppDIRAC.Lib import Conf
 from WebAppDIRAC.Lib.WebHandler import WebHandler, WebSocketHandler
 from WebAppDIRAC.Core.CoreHandler import CoreHandler
 from WebAppDIRAC.Core.StaticHandler import StaticHandler
-from WebAppDIRAC.Lib import Conf
+
+__RCSID__ = "$Id$"
 
 
 class HandlerMgr(object):
   __metaclass__ = DIRACSingleton
 
-  def __init__(self, baseURL=""):
+  def __init__(self, baseURL="/", sysService=None):
     self.__baseURL = baseURL.strip("/")
+    self.__sysService = sysService or []
+    if not isinstance(self.__sysService, list):
+      self.__sysService = self.__sysService.replace(' ', '').split(',')
     self.__routes = []
     self.__handlers = []
     self.__setupGroupRE = r"(?:/s:([\w-]*)/g:([\w.-]*))?"
     self.__shySetupGroupRE = r"(?:/s:(?:[\w-]*)/g:(?:[\w.-]*))?"
     self.log = gLogger.getSubLogger("Routing")
 
   def getPaths(self, dirName):
-    """
-    Get lists of paths for all installed and enabled extensions
+    """ Get lists of paths for all installed and enabled extensions
     """
     pathList = []
     for extName in CSGlobals.getCSExtensions():
       if extName.rfind("DIRAC") != len(extName) - 5:
         extName = "%sDIRAC" % extName
-      if extName == "WebAppDIRAC":
-        continue
```

Comment:
Why you removed this? This part the code is handling the extensions such as LHCbWebAppDIRAC.
Are you sure it will work without this lines?

Reply:
I don't remember actually. I agree that logic here not complete without this lines. Now its fixed.